### PR TITLE
Objects apigw API

### PR DESCRIPTION
**Pull request by brainstorm**
Adds endpoints to query the filemanager model from REST (API Gateway V2). Also:

* Refactors and promotes api-gw from SRM to top level.
* Fixes all workloads to use the new construct instead of custom ones per service.
* Adds AccessLog logGroup.
* Explores CloudMap at ApiGw level.

### CODE REVIEW

File: lib/workload/stateless/stacks/filemanager/deploy/lib/filemanager.ts [hunk 1]
```diff
@@ -101,5 +120,44 @@ export class Filemanager extends Stack {
...
+    const pool = new UserPool(this, "userPool", {
+      userPoolName: "OrcaBusUserPool",
+    });
+
+    // JWT authorizer
+    const authorizer = new HttpJwtAuthorizer(
+        id + "FilemanagerQueryApiAuthorizer",
+        `https://cognito-idp.${this.region}.amazonaws.com/`,
+        {
+          identitySource: ["meow"], //["$request.header.Authorization"],
```
Comment (andrewpatto) [inline lib/workload/stateless/stacks/filemanager/deploy/lib/filemanager.ts:145]: meow meow

Comment (brainstorm) [reply lib/workload/stateless/stacks/filemanager/deploy/lib/filemanager.ts:145]: Just wanted to `cdk synth`, so I threw the cat in there X"D

**Comment by victorskl** [comment]
will review today

**Comment by victorskl** [comment]
reviewing...

File: lib/workload/components/api-gateway/index.ts [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
+export class ApiGatewayConstruct extends Construct {
   private readonly _httpApi: HttpApi;
 
-  constructor(scope: Construct, id: string, props: SRMApiGatewayConstructProps) {
+  constructor(scope: Construct, id: string, props: ApiGatewayConstructProps) {
     super(scope, id);
 
     this._httpApi = new HttpApi(this, 'HttpApi', {
-      apiName: 'OrcaBusAPI-SequenceRunManager',
+      apiName: 'OrcaBusAPI-' + id,
```
Comment (victorskl) [inline lib/workload/components/api-gateway/index.ts:23]: Can we expose this `apiName` prop as `ApiGatewayConstructProps` and, allow set by the caller.

This CDK construct ID behaviour is context specific depends on its parent stack.

The prefix `OrcaBusAPI-` is fine, still. 

i.e.

```
apiName: 'OrcaBusAPI-' + props.apiName,
```

And expose `apiName` prop at the above interface `ApiGatewayConstructProps`.

File: lib/workload/stateless/stacks/filemanager/deploy/stack.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+      vpc: this.vpc,
+      host: this.host,
+      securityGroup: this.securityGroup,
+      ...props
+    });
+
+    const ApiGateway = new ApiGatewayConstruct(this, 'ApiGateway-'+props.stackName, {
+      region: this.region,
+      ...props,
+    });
```
Comment (victorskl) [inline lib/workload/stateless/stacks/filemanager/deploy/stack.ts:118]: For the construct ID, just 'ApiGateway' should do it. Then, we set the `apiName` prop like:

```
    const ApiGateway = new ApiGatewayConstruct(this, 'ApiGateway', {
      region: this.region,
      apiName: props.stackName
      ...props,
    });
```

File: lib/workload/stateless/stacks/metadata-manager/deploy/stack.ts [hunk 4]
```diff
@@ -80,11 +80,17 @@ export class MetadataManagerStack extends Stack {
...
 
     // (1)
-    const apiGW = new ApiGatewayConstruct(this, 'APIGW').httpApi;
+    const apiGW = new ApiGatewayConstruct(this, 'OrcabusAPI-MetadataManager', {
+      region: this.region,
+      cognitoUserPoolIdParameterName: 'YOUR_USER_POOL_ID_PARAMETER_NAME',
+      cognitoPortalAppClientIdParameterName: 'YOUR_PORTAL_APP_CLIENT_ID_PARAMETER_NAME',
+      cognitoStatusPageAppClientIdParameterName: 'YOUR_STATUS_PAGE_APP_CLIENT_ID_PARAMETER_NAME',
+      ...props,
+    });
```
Comment (victorskl) [inline lib/workload/stateless/stacks/metadata-manager/deploy/stack.ts:89]: Similarly. The set `apiName` prop like so:

```
    const apiGW = new ApiGatewayConstruct(this, 'ApiGateway', {
      region: this.region,
      cognitoUserPoolIdParameterName: 'YOUR_USER_POOL_ID_PARAMETER_NAME',
      cognitoPortalAppClientIdParameterName: 'YOUR_PORTAL_APP_CLIENT_ID_PARAMETER_NAME',
      cognitoStatusPageAppClientIdParameterName: 'YOUR_STATUS_PAGE_APP_CLIENT_ID_PARAMETER_NAME',
      apiName: 'MetadataManager',
      ...props,
    });
```

File: lib/workload/stateless/stacks/sequence-run-manager/deploy/stack.ts [hunk 5]
```diff
@@ -111,7 +111,7 @@ export class SequenceRunManagerStack extends Stack {
       timeout: Duration.seconds(28),
     });
 
-    const srmApi = new SRMApiGatewayConstruct(this, 'SRMApiGatewayConstruct', {
+    const srmApi = new ApiGatewayConstruct(this, 'SRMApiGatewayConstruct', {
```
Comment (victorskl) [inline lib/workload/stateless/stacks/sequence-run-manager/deploy/stack.ts:114]: And, similarly.

```
const srmApi = new ApiGatewayConstruct(this, 'ApiGateway', {
      region: this.region,
      apiName: 'SequenceRunManager',
      ...props,
    });
```

**Review by victorskl — APPROVED** [review]
LGTM. Minor comment/suggestion to attend.

**Pull request merged**